app/models/event.rb

```diff
@@ -0,0 +1,52 @@
+class Event < ActiveRecord::Base
+  after_save :add_organizer_as_participant
+
+  has_many :participated_event_relationships, class_name: 'UserEvent'
+  has_many :participants, class_name: 'User', through: :participated_event_relationships
+  belongs_to :organizer, class_name: 'User'
+  belongs_to :event_address
+
+  delegate :name, to: :organizer, prefix: :organizer
+  delegate :address, to: :event_address
+  delegate :coords, to: :event_address
+
+  validates :organizer, presence: true
+  validates :name, presence: true
+  validates :event_address, presence: true
+  validates :scheduled_at, presence: true
+  validates :scheduled_hours, numericality: {greater_than_or_equal_to: 0}, allow_nil: true
+  validates :capacity, numericality: {greater_than_or_equal_to: 0, only_interger: true}, allow_nil: true
+
+  scope :public_only, -> { where(is_private: false) }
+  scope :today, -> { where("scheduled_at >= ? and scheduled_at <= ?", Date.today.beginning_of_day, Date.today.end_of_day) }
+  scope :past, -> { where("scheduled_at < ?", Date.today.beginning_of_day) }
+  scope :upcoming, -> { where("scheduled_at > ?", Date.today.end_of_day) }
+
```

Comment:
For time-dependent scopes, consider using Rails' `Time.zone.today` instead of Ruby's `Date.today` to ensure the time zone is consistent with the configured application timezone. Alternatively, it may be more accurate to use the user's own time zone (perhaps saved in their preferences or figured out from their location?) and pass that time zone in as an argument to the scopes.